src/mscorlib/src/System/Text/StringBuilder.CoreCLR.cs

```diff
@@ -8,11 +8,86 @@ namespace System.Text
 {
     public partial class StringBuilder
     {
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        internal extern unsafe void ReplaceBufferInternal(char* newBuffer, int newLength);
+        // This function assumes that requiredLength will be less 
+        // than the max capacity of the StringBuilder
+        private int GetAllocationLength(int requiredLength)
+        {
+            int GetArrayLength()
```

Comment:
This is the same as `Capacity` property getter.

Reply:
Yep, there are  many things that can be cleaned up in this code...